### ISSUE TITLE
no-empty-interface : incorrectly defining problem when using Omit 

### ISSUE THREAD

**Issue by jaycenhorton** [typescript-eslint/typescript-eslint#585]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/no-empty-interface": "error"
  }
}
```

```TS
interface Trigger {
  shouldExecute: boolean;
  data: {
    mutation: string;
    [key: string]: any;
  };
  when: Date | number;
  key: string;
}
interface Schedule extends Omit<Trigger, 'shouldExecute'> {} 

```

**Expected Result**
This should not be an error, as the interface `Schedule` is not equal to `Trigger` as it is instead

```TS
interface Trigger {
  data: {
    mutation: string;
    [key: string]: any;
  };
  when: Date | number;
  key: string;
}
``` 

**Actual Result**

On the interface `Schedule`:

```
An interface declaring no members is equivalent to its supertype.eslint(@typescript-eslint/no-empty-interface)
```

**Additional Info**

```
2019-06-03T19:53:11.061Z eslint:cli CLI args: [ '--debug', '/home/ubuntu/dev/nori/nori-scheduler/src/tmp.ts' ]
2019-06-03T19:53:11.063Z eslint:cli Running on files
2019-06-03T19:53:11.066Z eslint:glob-utils Creating list of files to process.
2019-06-03T19:53:11.067Z eslint:ignored-paths baseDir = "/home/ubuntu/dev/nori"
2019-06-03T19:53:11.067Z eslint:ignored-paths addPatternRelativeToCwd:
  original = "/node_modules/*"
  cooked   = "/node_modules/*"
2019-06-03T19:53:11.068Z eslint:ignored-paths addPatternRelativeToCwd:
  original = "/bower_components/*"
  cooked   = "/bower_components/*"
2019-06-03T19:53:11.068Z eslint:ignored-paths addPatternRelativeToCwd:
  original = ".*"
  cooked   = ".*"
2019-06-03T19:53:11.068Z eslint:ignored-paths addPatternRelativeToCwd:
  original = "!../"
  cooked   = "!../"
2019-06-03T19:53:11.068Z eslint:ignored-paths Looking for ignore file in /home/ubuntu/dev/nori
2019-06-03T19:53:11.068Z eslint:ignored-paths Loaded ignore file /home/ubuntu/dev/nori/.eslintignore
2019-06-03T19:53:11.068Z eslint:ignored-paths Adding /home/ubuntu/dev/nori/.eslintignore
2019-06-03T19:53:11.068Z eslint:ignored-paths addPatternRelativeToIgnoreFile:
  original = ".next"
  cooked   = ".next"
2019-06-03T19:53:11.068Z eslint:ignored-paths addPatternRelativeToIgnoreFile:
  original = "__generated__"
  cooked   = "__generated__"
2019-06-03T19:53:11.069Z eslint:ignored-paths addPatternRelativeToIgnoreFile:
  original = "_next"
  cooked   = "_next"
2019-06-03T19:53:11.069Z eslint:ignored-paths addPatternRelativeToIgnoreFile:
  original = ".next"
  cooked   = ".next"
2019-06-03T19:53:11.069Z eslint:ignored-paths addPatternRelativeToIgnoreFile:
  original = "__generated__"
  cooked   = "__generated__"
2019-06-03T19:53:11.069Z eslint:ignored-paths addPatternRelativeToIgnoreFile:
  original = "_next"
  cooked   = "_next"
2019-06-03T19:53:11.070Z eslint:ignored-paths contains:
2019-06-03T19:53:11.070Z eslint:ignored-paths   target = "/home/ubuntu/dev/nori/nori-scheduler/src/tmp.ts"
2019-06-03T19:53:11.070Z eslint:ignored-paths   result = false
2019-06-03T19:53:11.070Z eslint:ignored-paths contains:
2019-06-03T19:53:11.070Z eslint:ignored-paths   target = "/home/ubuntu/dev/nori/nori-scheduler/src/tmp.ts"
2019-06-03T19:53:11.070Z eslint:ignored-paths   result = false
2019-06-03T19:53:11.070Z eslint:cli-engine Processing /home/ubuntu/dev/nori/nori-scheduler/src/tmp.ts
2019-06-03T19:53:11.070Z eslint:cli-engine Linting /home/ubuntu/dev/nori/nori-scheduler/src/tmp.ts
2019-06-03T19:53:11.071Z eslint:config Constructing config file hierarchy for /home/ubuntu/dev/nori/nori-scheduler/src
2019-06-03T19:53:11.071Z eslint:config Using .eslintrc and package.json files
2019-06-03T19:53:11.071Z eslint:config Loading /home/ubuntu/dev/nori/nori-scheduler/.eslintrc.js
2019-06-03T19:53:11.072Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/nori-scheduler/.eslintrc.js
2019-06-03T19:53:11.298Z eslint:plugins Loaded plugin @typescript-eslint (@typescript-eslint/eslint-plugin, version unknown) (from /home/ubuntu/dev/nori/node_modules/@typescript-eslint/eslint-plugin/dist/index.js)
2019-06-03T19:53:11.354Z eslint:plugins Loaded plugin import (eslint-plugin-import, version unknown) (from /home/ubuntu/dev/nori/node_modules/eslint-plugin-import/lib/index.js)
2019-06-03T19:53:11.355Z eslint:plugins Loaded plugin sort-imports-es6-autofix (eslint-plugin-sort-imports-es6-autofix, version unknown) (from /home/ubuntu/dev/nori/node_modules/eslint-plugin-sort-imports-es6-autofix/index.js)
2019-06-03T19:53:11.435Z eslint:plugins Loaded plugin graphql (eslint-plugin-graphql, version unknown) (from /home/ubuntu/dev/nori/node_modules/eslint-plugin-graphql/lib/index.js)
2019-06-03T19:53:11.462Z eslint:config-file Loading plugin:import/errors
2019-06-03T19:53:11.462Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-plugin-import/lib/index.js
2019-06-03T19:53:11.465Z eslint:config-file Loading plugin:import/warnings
2019-06-03T19:53:11.465Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-plugin-import/lib/index.js
2019-06-03T19:53:11.465Z eslint:config-file Loading plugin:import/typescript
2019-06-03T19:53:11.466Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-plugin-import/lib/index.js
2019-06-03T19:53:11.466Z eslint:config-file Loading plugin:require-path-exists/recommended
2019-06-03T19:53:11.467Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-plugin-require-path-exists/index.js
2019-06-03T19:53:11.483Z eslint:plugins Loaded plugin require-path-exists (eslint-plugin-require-path-exists, version unknown) (from /home/ubuntu/dev/nori/node_modules/eslint-plugin-require-path-exists/index.js)
2019-06-03T19:53:11.483Z eslint:config-file Loading prettier/@typescript-eslint
2019-06-03T19:53:11.483Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-prettier/@typescript-eslint.js
2019-06-03T19:53:11.484Z eslint:config-file Loading prettier
2019-06-03T19:53:11.484Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-prettier/index.js
2019-06-03T19:53:11.484Z eslint:config-file Loading plugin:@typescript-eslint/recommended
2019-06-03T19:53:11.484Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/@typescript-eslint/eslint-plugin/dist/index.js
2019-06-03T19:53:11.510Z eslint:config-file Loading airbnb-base
2019-06-03T19:53:11.510Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/index.js
2019-06-03T19:53:11.511Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/imports.js
2019-06-03T19:53:11.511Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/imports.js
2019-06-03T19:53:11.514Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/es6.js
2019-06-03T19:53:11.514Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/es6.js
2019-06-03T19:53:11.525Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/variables.js
2019-06-03T19:53:11.525Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/variables.js
2019-06-03T19:53:11.531Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/style.js
2019-06-03T19:53:11.531Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/style.js
2019-06-03T19:53:11.602Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/node.js
2019-06-03T19:53:11.602Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/node.js
2019-06-03T19:53:11.603Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/errors.js
2019-06-03T19:53:11.603Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/errors.js
2019-06-03T19:53:11.607Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/best-practices.js
2019-06-03T19:53:11.607Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/best-practices.js
2019-06-03T19:53:11.621Z eslint:config Using /home/ubuntu/dev/nori/nori-scheduler/.eslintrc.js
2019-06-03T19:53:11.621Z eslint:config Loading /home/ubuntu/dev/nori/.eslintrc.js
2019-06-03T19:53:11.621Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/.eslintrc.js
2019-06-03T19:53:11.666Z eslint:plugins Loaded plugin relay (eslint-plugin-relay, version unknown) (from /home/ubuntu/dev/nori/node_modules/eslint-plugin-relay/eslint-plugin-relay.js)
2019-06-03T19:53:11.667Z eslint:plugins Loaded plugin react-hooks (eslint-plugin-react-hooks, version unknown) (from /home/ubuntu/dev/nori/node_modules/eslint-plugin-react-hooks/index.js)
2019-06-03T19:53:11.668Z eslint:config-file Loading plugin:import/errors
2019-06-03T19:53:11.669Z eslint:config-file Loading plugin:import/warnings
2019-06-03T19:53:11.669Z eslint:config-file Loading plugin:relay/recommended
2019-06-03T19:53:11.669Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-plugin-relay/eslint-plugin-relay.js
2019-06-03T19:53:11.670Z eslint:config-file Loading plugin:require-path-exists/recommended
2019-06-03T19:53:11.670Z eslint:config-file Loading prettier/react
2019-06-03T19:53:11.670Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-prettier/react.js
2019-06-03T19:53:11.670Z eslint:config-file Loading prettier
2019-06-03T19:53:11.671Z eslint:config-file Loading airbnb
2019-06-03T19:53:11.671Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb/index.js
2019-06-03T19:53:11.671Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb/rules/react-a11y.js
2019-06-03T19:53:11.672Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb/rules/react-a11y.js
2019-06-03T19:53:11.758Z eslint:plugins Loaded plugin jsx-a11y (eslint-plugin-jsx-a11y, version unknown) (from /home/ubuntu/dev/nori/node_modules/eslint-plugin-jsx-a11y/lib/index.js)
2019-06-03T19:53:11.790Z eslint:plugins Loaded plugin react (eslint-plugin-react, version unknown) (from /home/ubuntu/dev/nori/node_modules/eslint-plugin-react/index.js)
2019-06-03T19:53:11.803Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb/rules/react.js
2019-06-03T19:53:11.804Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb/rules/react.js
2019-06-03T19:53:11.829Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/strict.js
2019-06-03T19:53:11.830Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/strict.js
2019-06-03T19:53:11.830Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/index.js
2019-06-03T19:53:11.830Z eslint:config-file Loading JS config file: /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/index.js
2019-06-03T19:53:11.830Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/imports.js
2019-06-03T19:53:11.831Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/es6.js
2019-06-03T19:53:11.831Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/variables.js
2019-06-03T19:53:11.831Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/style.js
2019-06-03T19:53:11.831Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/node.js
2019-06-03T19:53:11.831Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/errors.js
2019-06-03T19:53:11.831Z eslint:config-file Loading /home/ubuntu/dev/nori/node_modules/eslint-config-airbnb-base/rules/best-practices.js
2019-06-03T19:53:11.832Z eslint:config Using /home/ubuntu/dev/nori/.eslintrc.js
2019-06-03T19:53:11.834Z eslint:config-ops Using config from partial cache
2019-06-03T19:53:11.837Z eslint:config-ops Apply environment settings to config
2019-06-03T19:53:11.837Z eslint:config-ops Creating config for environment node
2019-06-03T19:53:11.837Z eslint:config-ops Creating config for environment es6
2019-06-03T19:53:11.837Z eslint:config-ops Creating config for environment shared-node-browser
2019-06-03T19:53:11.843Z eslint:linter Linting code for /home/ubuntu/dev/nori/nori-scheduler/src/tmp.ts (pass 1)
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.2.1 <3.5.0

YOUR TYPESCRIPT VERSION: 3.5.1

Please only submit bug reports when using the officially supported version.

=============
2019-06-03T19:53:11.862Z eslint:traverser Unknown node type "TSInterfaceDeclaration": Estimated visitor keys ["type","body","id","range","loc"]
2019-06-03T19:53:11.862Z eslint:traverser Unknown node type "TSInterfaceBody": Estimated visitor keys ["type","body","range","loc"]
2019-06-03T19:53:11.862Z eslint:traverser Unknown node type "TSPropertySignature": Estimated visitor keys ["type","optional","computed","key","typeAnnotation","initializer","readonly","static","export","range","loc"]
2019-06-03T19:53:11.862Z eslint:traverser Unknown node type "TSTypeAnnotation": Estimated visitor keys ["type","loc","range","typeAnnotation"]
2019-06-03T19:53:11.862Z eslint:traverser Unknown node type "TSTypeLiteral": Estimated visitor keys ["type","members","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSPropertySignature": Estimated visitor keys ["type","optional","computed","key","typeAnnotation","initializer","readonly","static","export","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSTypeAnnotation": Estimated visitor keys ["type","loc","range","typeAnnotation"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSStringKeyword": Estimated visitor keys ["type","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSIndexSignature": Estimated visitor keys ["type","parameters","range","loc","typeAnnotation"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSTypeAnnotation": Estimated visitor keys ["type","loc","range","typeAnnotation"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSAnyKeyword": Estimated visitor keys ["type","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSPropertySignature": Estimated visitor keys ["type","optional","computed","key","typeAnnotation","initializer","readonly","static","export","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSTypeAnnotation": Estimated visitor keys ["type","loc","range","typeAnnotation"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSBooleanKeyword": Estimated visitor keys ["type","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSPropertySignature": Estimated visitor keys ["type","optional","computed","key","typeAnnotation","initializer","readonly","static","export","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSTypeAnnotation": Estimated visitor keys ["type","loc","range","typeAnnotation"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSUnionType": Estimated visitor keys ["type","types","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSTypeReference": Estimated visitor keys ["type","typeName","typeParameters","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSNumberKeyword": Estimated visitor keys ["type","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSPropertySignature": Estimated visitor keys ["type","optional","computed","key","typeAnnotation","initializer","readonly","static","export","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSTypeAnnotation": Estimated visitor keys ["type","loc","range","typeAnnotation"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSStringKeyword": Estimated visitor keys ["type","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSInterfaceDeclaration": Estimated visitor keys ["type","body","id","range","loc","extends"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSInterfaceBody": Estimated visitor keys ["type","body","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSInterfaceHeritage": Estimated visitor keys ["type","expression","range","loc","typeParameters"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSTypeParameterInstantiation": Estimated visitor keys ["type","range","loc","params"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSTypeReference": Estimated visitor keys ["type","typeName","typeParameters","range","loc"]
2019-06-03T19:53:11.863Z eslint:traverser Unknown node type "TSLiteralType": Estimated visitor keys ["type","literal","range","loc"]
2019-06-03T19:53:12.039Z eslint:linter Generating fixed text for /home/ubuntu/dev/nori/nori-scheduler/src/tmp.ts (pass 1)
2019-06-03T19:53:12.039Z eslint:source-code-fixer Applying fixes
2019-06-03T19:53:12.039Z eslint:source-code-fixer shouldFix parameter was false, not attempting fixes
2019-06-03T19:53:12.040Z eslint:cli-engine Linting complete in: 975ms

/home/ubuntu/dev/nori/nori-scheduler/src/tmp.ts
   4:20  warning  Unexpected any. Specify a different type                          @typescript-eslint/no-explicit-any
  10:18  error    An interface declaring no members is equivalent to its supertype  @typescript-eslint/no-empty-interface

✖ 2 problems (1 error, 1 warning)

```

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.9.0` |
| `@typescript-eslint/parser`        | `1.9.0` |
| `TypeScript`                       | `3/5/1` |
| `ESLint`                           | `eslint@6.0.0-alpha.0` |
| `node`                             | `10.15.1` |
| `npm`                              | `6.9.0` |

**Comment by bradzacher** [comment]
```diff
-interface Schedule extends Omit<Trigger, 'shouldExecute'> {} 
+type Schedule = Omit<Trigger, 'shouldExecute'>;
```

The rule's error is not saying that `Schedule === Trigger`, it's saying that `Schedule === Omit<Trigger, 'shouldExecute'>`.

The solution to this is either use a `type` declaration, or add fields to the interface so it's not empty.
